src/CityOnPlanet.cpp

```diff
@@ -85,69 +86,179 @@ void CityOnPlanet::RemoveStaticGeomsFromCollisionSpace()
+			continue;
 		}
+
+		cityType.sizeDefs.emplace_back(std::move(radius));
 	}
-	assert(!models.empty());
-	Output("Got %d buildings of tag %s\n", static_cast<int>(models.size()), list->modelTagName);
-	list->buildings = new citybuilding_t[models.size()];
-	list->numBuildings = models.size();
-
-	int i = 0;
-	for (auto m = models.begin(), itEnd = models.end(); m != itEnd; ++m, i++) {
-		list->buildings[i].instIndex = i;
-		list->buildings[i].resolvedModel = *m;
-		list->buildings[i].idle = (*m)->FindAnimation("idle");
-		list->buildings[i].collMesh = (*m)->CreateCollisionMesh();
-		const Aabb &aabb = list->buildings[i].collMesh->GetAabb();
-		const double maxx = std::max(fabs(aabb.max.x), fabs(aabb.min.x));
-		const double maxy = std::max(fabs(aabb.max.z), fabs(aabb.min.z));
-		list->buildings[i].xzradius = sqrt(maxx * maxx + maxy * maxy);
-		Output(" - %s: %f\n", (*m)->GetName().c_str(), list->buildings[i].xzradius);
-	}
-	Output("End of buildings.\n");
+
+	Log::Verbose("\tLoaded {} city size definitions", numSizeDefs);
+
+	uint32_t numBuildingDefs = 0;
+
+	for (const auto &iter : buildingDefs.items()) {
+		BuildingType building = {};
+		numBuildingDefs++;
+
+		try {
+			LoadBuildingType(iter.key(), iter.value(), building);
+		} catch (std::exception &e) {
+			Log::Warning("Couldn't parse building def {}; error: {}", numBuildingDefs, e.what());
+			continue;
+		}
+
+		cityType.buildingTypes.emplace_back(std::move(building));
+	}
+
+	Log::Verbose("\tLoaded {} building definitions", numBuildingDefs);
+
+	s_cityFlavours.emplace_back(std::move(cityType));
+
+	Log::Info("\tCreated city definition flavour #{}", s_cityFlavours.size());
 }
 
 void CityOnPlanet::Init()
 {
 	PROFILE_SCOPED()
-	/* Resolve city model numbers since it is a bit expensive */
-	LookupBuildingListModels(&s_buildingList);
+
+	// Load all city definition configs
+	FileSystem::FileEnumerator iter(FileSystem::gameDataFiles, "configs/buildings/");
+	for (; !iter.Finished(); iter.Next()) {
+		const FileSystem::FileInfo &file = iter.Current();
+		std::string name = file.GetName();
+
+		if (!ends_with_ci(name, ".json")) {
+			continue;
+		}
+
+		try {
+			LoadCityFlavour(file);
+		} catch (std::exception &e) {
+			Log::Warning("CityOnPlanet: failed to parse city definition '{}': {}", name, e.what());
+		}
+	}
+
+	if (s_cityFlavours.empty()) {
+		Log::Error("CityOnPlanet: failed to load any city definitions, no cities will be generated!");
+	}
+
+	Graphics::RenderStateDesc rsd = {};
+	rsd.depthTest = false;
+	rsd.depthWrite = false;
+	rsd.primitiveType = Graphics::LINE_SINGLE;
+
+	s_debugMat.reset(Pi::renderer->CreateMaterial("vtxColor", Graphics::MaterialDescriptor(), rsd));
 }
 
 void CityOnPlanet::Uninit()
 {
-	delete[] s_buildingList.buildings;
+	s_cityFlavours.clear();
+	s_debugMat.reset();
 }
 
 // Need a reliable way to sort the models rather than using their address in memory we use their name which should be unique.
```

Comment:
the next two lines don't seem to be used anywhere

Reply:
Not only are those two functions unused but I think the ModelNameComparator struct is also unused. Thanks for catching that!